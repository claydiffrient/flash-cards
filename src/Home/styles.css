.root {
  padding: 20px;
  background-color: rgba(0, 0, 0, 0.8);
  height: auto;
  position: fixed;
  left: 5vw;
  right: 5vw;
  top: 5vh;
  bottom: 5vh;
  display: inline;
}

.content {
  border: 1px dotted red;
  text-align: left;
}
