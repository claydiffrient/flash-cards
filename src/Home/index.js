import React, { Component } from "react";
import Heading from "@instructure/ui-core/lib/components/Heading";
import Text from "@instructure/ui-core/lib/components/Text";
import Container from "@instructure/ui-core/lib/components/Container";
import List, { ListItem } from "@instructure/ui-core/lib/components/List";
import Button from "@instructure/ui-core/lib/components/Button";
import Modal, {
  ModalBody,
  ModalHeader
} from "@instructure/ui-core/lib/components/Modal";
import DeckPicker from "../DeckPicker";

import themeable from "@instructure/ui-themeable";
import styles from "./styles.css";
import theme from "./theme.js";

class Home extends Component {
  state = {
    deckPickerModalOpen: false
  };

  openDeckPickerModal = () => this.setState({ deckPickerModalOpen: true });
  closeDeckPickerModal = () => this.setState({ deckPickerModalOpen: false });

  render() {
    return (
      <div className={styles.root}>
        <Heading level="h1" color="primary-inverse">
          Flash Cards
        </Heading>

        <Container as="div" textAlign="start">
          <Heading color="primary-inverse" level="h1" margin="0 0 x-small">
            What is This?
          </Heading>
          <Text as="p" color="primary-inverse">
            Flash Cards is a web app being developed to help with learning
            things. {"It's"} being developed by Clay Diffrient as a project for
            the EDPS 6560 Mulitimedia Learning class in the pursuit of a M.Ed in
            Instructional Design and Educational Technology.
          </Text>
          <Heading color="primary-inverse" margin="0 0 x-small">
            What Will It Do?
          </Heading>
          <Text as="p" color="primary-inverse">
            I have somewhat grand aspirations for what it will do eventually,
            but for now I hope to do the following things:
          </Text>
          <List>
            <ListItem>
              <Text color="primary-inverse">
                [DONE] Show a flash card and flip to the other side via user
                interaction.
              </Text>
            </ListItem>
            <ListItem>
              <Text color="primary-inverse">
                [DONE] Show a flash card and flip to the other side after a
                given time.
              </Text>
            </ListItem>
            <ListItem>
              <Text color="primary-inverse">
                Track correct vs. incorrect answers by user input
              </Text>
            </ListItem>
            <ListItem>
              <Text color="primary-inverse">
                Summarize stats at the end of a session
              </Text>
            </ListItem>
            <ListItem>
              <Text color="primary-inverse">
                [DONE] Allow import/export of flash cards in a file format
                (likely JSON)
              </Text>
            </ListItem>
          </List>
        </Container>
        <Container as="div" textAlign="center" margin="medium 0 0 0">
          <Button onClick={this.openDeckPickerModal}>Start</Button>
        </Container>
        <Modal
          label="Deck Selector"
          closeButtonLabel="Close"
          open={this.state.deckPickerModalOpen}
          applicationElement={() => document.getElementById("main")}
          onDismiss={this.closeDeckPickerModal}
          size="medium"
        >
          <ModalHeader>
            <Heading>Choose a Deck of Cards</Heading>
          </ModalHeader>
          <ModalBody>
            <DeckPicker />
          </ModalBody>
        </Modal>
      </div>
    );
  }
}

export default themeable(theme, styles)(Home);
