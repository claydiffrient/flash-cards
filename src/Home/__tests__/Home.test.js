import React from "react";
import { MemoryRouter } from "react-router";
import { mount } from "enzyme";
import Home from "../index";

it("renders", () => {
  const wrapper = mount(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );
  expect(wrapper.find("Home")).toMatchSnapshot();
});
