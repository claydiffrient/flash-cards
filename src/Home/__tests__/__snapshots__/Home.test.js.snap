// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Home>
  <div
    className="Home__root__2fUWydo"
  >
    <Heading
      border="none"
      color="primary-inverse"
      ellipsis={false}
      level="h1"
    >
      <Container
        as="h1"
        className="_3BZ02Bg _3eNxhb- _3VqD3Wt"
        display={null}
      >
        <h1
          className="_3BZ02Bg _3eNxhb- _3VqD3Wt"
          style={
            Object {
              "margin": null,
              "padding": null,
            }
          }
        >
          Flash Cards
        </h1>
      </Container>
    </Heading>
    <Container
      as="div"
      display={null}
      textAlign="start"
    >
      <div
        className="_2iJkRW"
        style={
          Object {
            "margin": null,
            "padding": null,
          }
        }
      >
        <Heading
          border="none"
          color="primary-inverse"
          ellipsis={false}
          level="h1"
          margin="0 0 x-small"
        >
          <Container
            as="h1"
            className="_3BZ02Bg _3eNxhb- _3VqD3Wt"
            display={null}
            margin="0 0 x-small"
          >
            <h1
              className="_3BZ02Bg _3eNxhb- _3VqD3Wt"
              style={
                Object {
                  "margin": "0 0 0.5rem",
                  "padding": null,
                }
              }
            >
              What is This?
            </h1>
          </Container>
        </Heading>
        <Text
          as="p"
          color="primary-inverse"
          letterSpacing="normal"
          size="medium"
        >
          <p
            className="_3jLMUSh _3fy1EZS _2WOMSfA _2G9WT0d"
          >
            Flash Cards is a web app being developed to help with learning things. 
            It's
             being developed by Clay Diffrient as a project for the EDPS 6560 Mulitimedia Learning class in the pursuit of a M.Ed in Instructional Design and Educational Technology.
          </p>
        </Text>
        <Heading
          border="none"
          color="primary-inverse"
          ellipsis={false}
          level="h2"
          margin="0 0 x-small"
        >
          <Container
            as="h2"
            className="_3BZ02Bg _1t9nPA1 _3VqD3Wt"
            display={null}
            margin="0 0 x-small"
          >
            <h2
              className="_3BZ02Bg _1t9nPA1 _3VqD3Wt"
              style={
                Object {
                  "margin": "0 0 0.5rem",
                  "padding": null,
                }
              }
            >
              What Will It Do?
            </h2>
          </Container>
        </Heading>
        <Text
          as="p"
          color="primary-inverse"
          letterSpacing="normal"
          size="medium"
        >
          <p
            className="_3jLMUSh _3fy1EZS _2WOMSfA _2G9WT0d"
          >
            I have somewhat grand aspirations for what it will do eventually, but for now I hope to do the following things:
          </p>
        </Text>
        <List
          as="ul"
          delimiter="none"
          margin="none"
          size="medium"
          variant="default"
        >
          <Container
            as="ul"
            className="_2j92CzO QuihDiE _1eZWDls"
            display={null}
            margin="none"
          >
            <ul
              className="_2j92CzO QuihDiE _1eZWDls"
              style={
                Object {
                  "margin": "0",
                  "padding": null,
                }
              }
            >
              <ListItem
                as="ul"
                delimiter="none"
                key=".0"
                size="medium"
                variant="default"
              >
                <Container
                  as="li"
                  className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                  display={null}
                >
                  <li
                    className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                    style={
                      Object {
                        "margin": null,
                        "padding": null,
                      }
                    }
                  >
                    <Text
                      as="span"
                      color="primary-inverse"
                      letterSpacing="normal"
                      size="medium"
                    >
                      <span
                        className="_3jLMUSh _3fy1EZS _2WOMSfA _2G9WT0d"
                      >
                        [DONE] Show a flash card and flip to the other side via user interaction.
                      </span>
                    </Text>
                  </li>
                </Container>
              </ListItem>
              <ListItem
                as="ul"
                delimiter="none"
                key=".1"
                size="medium"
                variant="default"
              >
                <Container
                  as="li"
                  className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                  display={null}
                >
                  <li
                    className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                    style={
                      Object {
                        "margin": null,
                        "padding": null,
                      }
                    }
                  >
                    <Text
                      as="span"
                      color="primary-inverse"
                      letterSpacing="normal"
                      size="medium"
                    >
                      <span
                        className="_3jLMUSh _3fy1EZS _2WOMSfA _2G9WT0d"
                      >
                        [DONE] Show a flash card and flip to the other side after a given time.
                      </span>
                    </Text>
                  </li>
                </Container>
              </ListItem>
              <ListItem
                as="ul"
                delimiter="none"
                key=".2"
                size="medium"
                variant="default"
              >
                <Container
                  as="li"
                  className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                  display={null}
                >
                  <li
                    className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                    style={
                      Object {
                        "margin": null,
                        "padding": null,
                      }
                    }
                  >
                    <Text
                      as="span"
                      color="primary-inverse"
                      letterSpacing="normal"
                      size="medium"
                    >
                      <span
                        className="_3jLMUSh _3fy1EZS _2WOMSfA _2G9WT0d"
                      >
                        Track correct vs. incorrect answers by user input
                      </span>
                    </Text>
                  </li>
                </Container>
              </ListItem>
              <ListItem
                as="ul"
                delimiter="none"
                key=".3"
                size="medium"
                variant="default"
              >
                <Container
                  as="li"
                  className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                  display={null}
                >
                  <li
                    className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                    style={
                      Object {
                        "margin": null,
                        "padding": null,
                      }
                    }
                  >
                    <Text
                      as="span"
                      color="primary-inverse"
                      letterSpacing="normal"
                      size="medium"
                    >
                      <span
                        className="_3jLMUSh _3fy1EZS _2WOMSfA _2G9WT0d"
                      >
                        Summarize stats at the end of a session
                      </span>
                    </Text>
                  </li>
                </Container>
              </ListItem>
              <ListItem
                as="ul"
                delimiter="none"
                key=".4"
                size="medium"
                variant="default"
              >
                <Container
                  as="li"
                  className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                  display={null}
                >
                  <li
                    className="_1dyDTaI undefined _1jLfonx _9PzDC58"
                    style={
                      Object {
                        "margin": null,
                        "padding": null,
                      }
                    }
                  >
                    <Text
                      as="span"
                      color="primary-inverse"
                      letterSpacing="normal"
                      size="medium"
                    >
                      <span
                        className="_3jLMUSh _3fy1EZS _2WOMSfA _2G9WT0d"
                      >
                        [DONE] Allow import/export of flash cards in a file format (likely JSON)
                      </span>
                    </Text>
                  </li>
                </Container>
              </ListItem>
            </ul>
          </Container>
        </List>
      </div>
    </Container>
    <Container
      as="div"
      display={null}
      margin="medium 0 0 0"
      textAlign="center"
    >
      <div
        className="_6GWxrTR"
        style={
          Object {
            "margin": "1.5rem 0 0 0",
            "padding": null,
          }
        }
      >
        <Button
          as="button"
          buttonRef={[Function]}
          fluidWidth={false}
          margin="0"
          onClick={[Function]}
          size="medium"
          type="button"
          variant="default"
        >
          <Container
            aria-disabled={null}
            as="button"
            className="_922amOO _16-3B-L _3Gg19qR"
            display={null}
            elementRef={[Function]}
            margin="0"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <button
              aria-disabled={null}
              className="_922amOO _16-3B-L _3Gg19qR"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "margin": "0",
                  "padding": null,
                }
              }
              tabIndex="0"
              type="button"
            >
              <span
                className="_2CUXt80"
              >
                Start
              </span>
            </button>
          </Container>
        </Button>
      </div>
    </Container>
    <Modal
      contentRef={[Function]}
      defaultFocusElement={null}
      insertAt="bottom"
      label="Deck Selector"
      liveRegion={null}
      mountNode={null}
      onClose={[Function]}
      onDismiss={[Function]}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      onOpen={[Function]}
      open={false}
      shouldCloseOnDocumentClick={true}
      shouldReturnFocus={true}
      size="medium"
      transition="fade"
    >
      <Portal
        insertAt="bottom"
        mountNode={null}
        onClose={[Function]}
        onOpen={[Function]}
        open={false}
      />
    </Modal>
  </div>
</Home>
`;
